tiatoolbox/models/architecture/sccnn.py

```diff
@@ -0,0 +1,360 @@
+            )
+            return nn.ModuleDict(module_dict)
+
+        module_dict = OrderedDict()
+        module_dict["l1"] = conv_act_branch(num_input_channels, 30, 2)
+        module_dict["pool1"] = nn.MaxPool2d(2, padding=0)
+        module_dict["l2"] = conv_act_branch(30, 60, 2)
+        module_dict["pool2"] = nn.MaxPool2d(2, padding=0)
+        module_dict["l3"] = conv_act_branch(60, 90, 3)
+        module_dict["l4"] = conv_act_branch(90, 1024, 5)
+        module_dict["dropout1"] = nn.Dropout2d(p=0.5)
+        module_dict["l5"] = conv_act_branch(1024, 512, 1)
+        module_dict["dropout2"] = nn.Dropout2d(p=0.5)
+        module_dict["sc"] = sc(512, 3)
+
+        self.layer = nn.ModuleDict(module_dict)
+
+    def forward(self, input_tensor: torch.Tensor):  # skipcq: PYL-W0221
+        """Logic for using layers defined in init.
+
+        This method defines how layers are used in forward operation.
+
+        Args:
+            input_tensor (torch.Tensor):
+                Input images, the tensor is in the shape of NCHW.
+
+        Returns:
+            torch.Tensor:
+                Output map for cell detection. Peak detection should be applied
+                to this output for cell detection.
+
+        """
+
+        def conv_act_branch(
+            layer: torch.nn.ModuleDict, in_tensor: torch.Tensor
+        ) -> torch.Tensor:
+            """Applies Convolution and Activation to the input tensor.
+
+            Args:
+                layer (torch.nn.ModuleDict):
+                    Torch layer as ModuleDict.
+                in_tensor (torch.Tensor):
+                    Input Tensor.
+            Returns:
+                torch.Tensor:
+                    Torch Tensor after applying convolution and activation.
+
+            """
+            return layer["conv1"](in_tensor)
+
+        def sc1_layer(
+            layer: torch.nn.ModuleDict,
+            in_tensor: torch.Tensor,
+            out_height: int = 13,
+            out_width: int = 13,
+        ) -> tuple:
+            """Spatially constrained layer 1.
+
+            Estimates row, column and height for sc2 layer mapping.
+
+            Args:
+                layer (torch.nn.ModuleDict):
+                    Torch layer as ModuleDict.
+                in_tensor (torch.Tensor):
+                    Input Tensor.
+                out_height (int):
+                    Output height.
+                out_width (int):
+                    Output Width
+
+            Returns:
+                tuple of :class:`torch.Tensor`:
+                    Row, Column and height estimates used for sc2 mapping.
+
+            """
+            sigmoid = layer["conv1"](in_tensor)
+            sigmoid0 = sigmoid[:, 0:1, :, :] * (out_height - 1)
+            sigmoid1 = sigmoid[:, 1:2, :, :] * (out_width - 1)
+            sigmoid2 = sigmoid[:, 2:3, :, :]
+            return sigmoid0, sigmoid1, sigmoid2
+
+        def sc2layer(network, sc1_0, sc1_1, sc1_2) -> torch.Tensor:
```

Comment:
Inconsistent naming with layer 1. Also unclear from the names what the 1 and 2 means.

Reply:
Fixed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/434/commits/6dcf8731a6dc27caa2cafd10ed24502253ecb530